src/test/java/com/connexta/ingest/service/impl/IngestServiceImplUnitTests.java

```diff
@@ -0,0 +1,360 @@
+    final String fileName = "fileName";
+    final Throwable throwable = mock(StoreException.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName)).thenThrow(throwable);
+
+    // when
+    final Throwable thrown =
+        assertThrows(
+            Throwable.class,
+            () ->
+                ingestService.ingest(
+                    fileSize, mimeType, inputStream, fileName, 1L, mock(InputStream.class)));
+
+    // then
+    assertThat(thrown, is(throwable));
+  }
+
+  /* END ingest StoreClient tests */
+
+  /* START ingest MetacardStorageAdaptor tests */
+
+  @Test
+  public void testIngestWhenMetacardStorageAdaptorThrowsStoreMetacardException() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final long metacardFileSize = 1L;
+    final InputStream metacardInputStream = mock(InputStream.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName))
+        .thenReturn(new URI("http://localhost:9041/mis/product/1234"));
+
+    final StoreMetacardException storeMetacardException = mock(StoreMetacardException.class);
+    doThrow(storeMetacardException)
+        .when(mockMetacardStorageAdaptor)
+        .store(eq(metacardFileSize), eq(METACARD_MEDIA_TYPE), eq(metacardInputStream), anyString());
+
+    // when
+    final Throwable thrown =
+        assertThrows(
+            Throwable.class,
+            () ->
+                ingestService.ingest(
+                    fileSize,
+                    mimeType,
+                    inputStream,
+                    fileName,
+                    metacardFileSize,
+                    metacardInputStream));
+
+    // then
+    assertThat(thrown, is(storeMetacardException));
+  }
+
+  @Test
+  public void testIngestWhenMetacardStorageAdaptorThrowsThrowable() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final long metacardFileSize = 1L;
+    final InputStream metacardInputStream = mock(InputStream.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName))
+        .thenReturn(new URI("http://localhost:9041/mis/product/1234"));
+
+    final Throwable throwable = mock(StoreException.class);
```

Comment:
The duplicated code in the class makes it harder to see how one test is different from another. This 19 line block of code is repeated twice. First starting at line 152 and then again starting at line 193:

```
    // given
    final IngestService ingestService =
        new IngestServiceImpl(
            mockStoreClient,
            mockMetacardStorageAdaptor,
            "http://localhost:9040/ingest/",
            mockTransformClient);

    final Long fileSize = 1L;
    final String mimeType = "mimeType";
    final InputStream inputStream = mock(InputStream.class);
    final String fileName = "fileName";
    final long metacardFileSize = 1L;
    final InputStream metacardInputStream = mock(InputStream.class);
    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName))
        .thenReturn(new URI("http://localhost:9041/mis/product/1234"));

    final StoreMetacardException storeMetacardException = mock(StoreMetacardException.class);
```
These 12 lines are duplicated 5 times:

```
            Throwable.class,
            () ->
                ingestService.ingest(
                    fileSize,
                    mimeType,
                    inputStream,
                    fileName,
                    metacardFileSize,
                    metacardInputStream));

    // then
    assertThat(thrown, is(storeMetacardException));
```

The important differences get lost in the many lines of identical or nearly identical code.

Reply:
I prefer to keep it explicit and inlined.